### DISCUSSION TITLE
Is viper created by the singleton type,the singleton type in all folder or in one folder? I'm some confused....

### DISCUSSION THREAD

**Discussion by Juminiy** [spf13/viper#1342] [Q&A]
I'm a new spf13/viper, but occurs some problem. 
First, I create a `cli-x.yml` in cli-x (the root folder) , and I set the viper config in cmd/root.go ,but when I want use the `cli-x.yml` in connect/mongo.go , It can't be used.
The folder structure is 
cli-x (root folder)
- cmd 
  - root.go (where I config viper)
- connect
  - mongo.go (where I want viper's config)
- cli-x.yml (where the config file)
the repo is https://github.com/Juminiy/cli-x

**Comment by sagikazarmark** [comment]
Based on the code [here](https://github.com/Juminiy/cli-x/blob/main/connect/mongo.go#L16) it seems you are trying to get config out of Viper before it's initialized. Try to use Viper from inside our commands.

**Reply by Juminiy** [reply]
already solved ,thanks a lot !
